frontend/src/store/store-types.ts

```diff
@@ -84,6 +84,7 @@ export type PartialMainTask = Partial<MainTask>;
  * The type of user settings.
  */
 export type Settings = {
+  readonly canvasCalendar: string | null;
```

Comment:
The type should be `string | null | undefined` because some db objects might not have this field. Since you are using `!= null` check, you automatically get the `undefined` case handled.
However, let's be precise here to avoid future confusion. You could either do `readonly canvasCalendar: string | null | undefined;` or eliminate the `undefined` case in firebase listener.

Reply:
Okay, changed the type to be `string | null | undefined`